web/src/markdown.ts

```diff
@@ -494,47 +598,84 @@ function handleTimestamp(time) {
     return `<time datetime="${escaped_isotime}">${escaped_time}</time>`;
 }
 
-function handleStream({stream_name, get_stream_by_name, stream_hash}) {
+function handleStream({
+    stream_name,
+    get_stream_by_name,
+    stream_hash,
+}: {
+    stream_name: string;
+    get_stream_by_name: (stream_name: string) =>
+        | {
+              stream_id: number;
+              name: string;
+          }
+        | undefined;
+    stream_hash: (stream_id: number) => string;
+}): string | undefined {
     const stream = get_stream_by_name(stream_name);
     if (stream === undefined) {
         return undefined;
     }
     const href = stream_hash(stream.stream_id);
-    return `<a class="stream" data-stream-id="${_.escape(stream.stream_id)}" href="/${_.escape(
-        href,
-    )}">#${_.escape(stream.name)}</a>`;
+    return `<a class="stream" data-stream-id="${_.escape(
+        stream.stream_id.toString(),
+    )}" href="/${_.escape(href)}">#${_.escape(stream.name)}</a>`;
 }
 
-function handleStreamTopic({stream_name, topic, get_stream_by_name, stream_topic_hash}) {
+function handleStreamTopic({
+    stream_name,
+    topic,
+    get_stream_by_name,
+    stream_topic_hash,
+}: {
+    stream_name: string;
+    topic: string;
+    get_stream_by_name: (stream_name: string) =>
+        | {
+              stream_id: number;
+              name: string;
+          }
+        | undefined;
+    stream_topic_hash: (stream_id: number, topic: string) => string;
+}): string | undefined {
     const stream = get_stream_by_name(stream_name);
     if (stream === undefined || !topic) {
         return undefined;
     }
     const href = stream_topic_hash(stream.stream_id, topic);
     const text = `#${stream.name} > ${topic}`;
     return `<a class="stream-topic" data-stream-id="${_.escape(
-        stream.stream_id,
+        stream.stream_id.toString(),
     )}" href="/${_.escape(href)}">${_.escape(text)}</a>`;
 }
 
-function handleTex(tex, fullmatch) {
+function handleTex(tex: string, fullmatch: string): string {
     try {
         return katex.renderToString(tex);
     } catch (error) {
-        if (error.message.startsWith("KaTeX parse error")) {
+        if (error instanceof katex.ParseError) {
             // TeX syntax error
             return `<span class="tex-error">${_.escape(fullmatch)}</span>`;
         }
-        throw new Error(error.message);
+        throw error;
```

Comment:
This is slightly different than what we were doing before, but is similar to how other typescript files handle things like this

Reply:
nevermind apparently I can't actually do `instanceof katex.ParseError`